### ISSUE TITLE
Switching to content-driven framework [Suggestion]

### ISSUE THREAD

**Issue by bartosz-skejcik** [Buzzpy/Dev-Encyclopedia#38]
I'm wandering. Why don't we switch to, for example, Astro.js? It's a framework built exactly for the type of websites that Dev Encyclopedia is.
 
 I'd be more than happy to make the first move in switching to Astro.js. I could write all the content that there is now and leave a detailed documentation for adding new things
 
 [Here](https://astro.build/) is a link to the Astor website where you can read more about it.
 
 UPDATE:
 I made the switch already :wink: [here](https://github.com/bartosz-skejcik/Dev-Encyclopedia) is the converted site. Tell me if you want to go through with it :smile: 

The CI/CD will work with Astor because when building the project it outputs clean HTML, CSS and JS 😃

We can also improve the functionality further because we can add a React component for the modal so it's easier to manage, cause for now I'm using the data.json that @4rnv provided in his https://github.com/Buzzpy/Dev-Encyclopedia/discussions/29

**Comment by Brajo280603** [comment]
I wholeheartedly agree with @DanielAndresCP

**Comment by bartosz-skejcik** [comment]
That's what I'm talking about! Hahahah

The issue is that no one has the power to make this live except from @Buzzpy

**Comment by Himanshuch8055** [comment]
I fully agree with @bartosz-skejcik. This is extremely beneficial for scaling up the application.

**Comment by Buzzpy** [comment]
Sorry for any delays people! I'm pretty new to "maintaining" open-source projects too so please excuse me for any issues that may occur due to my... dumbness?

Anyhow, I think we should all agree on one thing: Using a framework like Astro or, integrating with a database where users can enter data by themselves. Or if's really an option, Markdown files? 

I'm more aligned to using Astro at the moment but I'd really love to know your opinions. Thanks for being patient! 🚀
@DanielAndresCP @Brajo280603 @pikhosh @RayMathew @bartosz-skejcik 

Related to issues #43 , #30, #18 #3 and #15

**Comment by RayMathew** [comment]
Astro itself uses markdown files. Or at least, that's what I understood from the documentation. So I'm also favoring it...

I don't think it's a good idea for users to be able to enter data themselves. Even if it's time consuming, there has to be strict quality control on what people enter. 

My suggestion would be to switch to Astro and then users can create markdowns as merge requests.

**Comment by bartosz-skejcik** [comment]
> Astro itself uses markdown files. Or at least, that's what I understood from the documentation. So I'm also favoring it...
> 
> I don't think it's a good idea for users to be able to enter data themselves. Even if it's time consuming, there has to be strict quality control on what people enter.
> 
> My suggestion would be to switch to Astro and then users can create markdowns as merge requests.

That's what I've been saying for the pas 2 days :smile:, Astro is pretty simple and easy to learn, so everybody would be able to contribute. I'm still wondering about the markdown files. As I can see the list of cards / concepts is pretty long so adding a file for each concept is just straight up dumb.

What if we instead create files for categories and just separate each contept in a file with a header os smth. like this:

```
# Concept title
concept desciption
<img src="some shit" alt="other shit" />

# Concept no. 2 title
concept no2. desciption
<img src="some shit 2" alt="other shit 2" />

...
```

**Comment by Buzzpy** [comment]
Great then let's go with Astro!

I didn't have time to look at the Astro docs yet, I will get into that soon. And about the markdown files, the categorical thing looks good but would that by any chance affect the structure of the website? I guess it won't, just wanted to make sure.

Thanks a lot for your contributions, really appreciate it.

**Comment by ManorSailor** [comment]
I’ve been following the repository since yesterday and have a few suggestions regarding the current workflow. However, if @Buzzpy has decided to go with Astro, that’s fine.

I haven’t worked with Astro myself, but from a quick look at the docs, it seems to generate static HTML from content provided through markdown files. Please correct me if I’m wrong.

> What if we create files for categories and separate each concept with a header or something?

@bartosz-skejcik, I suggest keeping content files separate. This approach makes it easier to extend and modify files as needed. While having one file might reduce the footprint, it could lead to Git conflicts when making changes, such as adding new terms or fixing grammar.

**Comment by Buzzpy** [comment]
> I’ve been following the repository since yesterday and had a few suggestions regarding the current workflow. However, if @Buzzpy has decided to go with Astro, that’s fine.
> 
> I haven’t worked with Astro myself, but from a quick look at the docs, it seems to generate static HTML from content provided through markdown files. Please correct me if I’m wrong.
> 
> > What if we create files for categories and separate each concept with a header or something?
> 
> @bartosz-skejcik, I suggest keeping content files separate. This approach makes it easier to extend and modify files as needed. While having one file might reduce the footprint, it could lead to Git conflicts when making changes, such as adding new terms or fixing grammar.

Thanks for your suggestion. But is OK to have about 150+ markdown files on this site? Would it affect the performance of the website? I'm a novice in web development so please bear with me and explain 🙏

**Comment by ManorSailor** [comment]
> Thanks for your suggestion. But is OK to have about 150+ markdown files on this site? Would it affect the performance of the website? I'm a novice in web development so please bear with me and explain 🙏

It's okay :). As for your query, No! It won't be an issue if we are using a framework like Astro. It'll generate the HTML from content or data files (markdown or json) at build time. Astro by default uses SSG (Static-Site Generation), which ensures everything is pre built waiting to be served to the user. With that said, I'm not entirely sure how it happens, but this is the gist of it. Maybe someone more knowledgeable can chime in

**Comment by Buzzpy** [comment]
Thanks @ManorSailor !

@Hussseinkizz introduced another simpler, yet effective framework called [Z.js](https://www.npmjs.com/package/z-js-framework). It's pretty new and not as popular as Astro, but after all, it's simple.

He has built a demo (minor changes have to be implemented, just for viewing) - [https://devpedia.netlify.app/](https://devpedia.netlify.app/)

Once implemented, contributors would just have to edit a single [JSON file](https://github.com/Hussseinkizz/Dev-Encyclopedia/blob/main/content/categories/all/data.json), which will be reviewed and then merged. 

View @Husseinkizz 's PR: [https://github.com/Buzzpy/Dev-Encyclopedia/pull/53](https://github.com/Buzzpy/Dev-Encyclopedia/pull/53)

Any comments? @RayMathew @bartosz-skejcik @Brajo280603 @DanielAndresCP

**Comment by ManorSailor** [comment]
@Buzzpy 

Whether be it Astro or Z.js or just vanilla JS. Astro brings the build step & exploring new concepts, Z.js & VJS (VanillaJS) bring the single source of truth issue, prone to git conflicts, for data. 

The point I'm trying to make is: Each approach comes with its own set of pros and cons.

With that said, I believe it's high time a decision is made on which stack we are going to migrate to. Only you can make that decision since it's your project, you define the path. 

Don't get me wrong, I understand how overwhelming this must be for you. Managing OSS projects is no easy feat. There tons of people proposing different solutions, features, & ideas. It's hard to keep up - wouldn't be surprising to say even unfeasible to implement some. 

I hope I wasn't too pushy (⁠٥⁠↼⁠_⁠↼⁠)

**Comment by RayMathew** [comment]
I checked the demo @Hussseinkizz built, and also checked Z.js documentation and reviews. 

With full respect @Hussseinkizz, I think it should be disclosed to everyone that you are the primary author of Z.js, and it is yet to get a decent number of reviews and community support. If we have to resolve bugs with the framework we would all be mostly relying on just you. Unless @Buzzpy is ok with that dependency, I would suggest going with a more known and tested framework, like Astro.

I hope you don't mind me saying that, I'm in great admiration of the work you've done to build a framework, and I'd aspire to be able to do so myself.


On a separate note, I don't understand the logic of putting all the data in a single JSON file? As it grows, it will take longer and longer to load, so the website will be blank till it does. Whereas individual files will load quicker, if not all at once.

**Comment by Ibrahim-Faisal15** [comment]
So when should the transition to the framework should begin or it has already begun?

**Comment by Hussseinkizz** [comment]
> @Buzzpy
> 
> Whether be it Astro or Z.js or just vanilla JS. Astro brings the build step & exploring new concepts, Z.js & VJS (VanillaJS) bring the single source of truth issue, prone to git conflicts, for data.
> 
> The point I'm trying to make is: Each approach comes with its own set of pros and cons.
> 
> With that said, I believe it's high time a decision is made on which stack we are going to migrate to. Only you can make that decision since it's your project, you define the path.
> 
> Don't get me wrong, I understand how overwhelming this must be for you. Managing OSS projects is no easy feat. There tons of people proposing different solutions, features, & ideas. It's hard to keep up - wouldn't be surprising to say even unfeasible to implement some.
> 
> I hope I wasn't too pushy (⁠٥⁠↼⁠_⁠↼⁠)

I would like to know what you mean by "bring the single source of truth issue, prone to git conflicts, for data." as this issue I addressed in my pr notes that we could have different json files separated by categories in that we only load one per a time, though ultimately a choice has to be made, who will be able to add to the collection, a normal dev or a contributor, that's should there be an interface for one posting a new entry (requires db server) or just people in some way either adding via markdown, json file etc...

**Comment by Hussseinkizz** [comment]
Ultimately much as it doesn't settle well, I recommend we get some kind of db to allow anyone to post a new entry that gets reviewed and then listed along others, people could even comment or upvote these to improve quality automatically or give augmentative info via commentary apart from the original author's perspective... of course even for a db, it should be as simple as possible: json server, pocket base, sqliite or others could be my go to!

**Comment by Hussseinkizz** [comment]
And with a backend, then suddenly any frontend including vanilla can do, modern frameworks have one trap, they have too much magic over the native web and therefore to maintain that magic feel they have to keep bringing more magic, ask a question will the framework you choose be around and still relatable in 10 years from now? or you will keep on having to rewrite the site just to fit the evolution? these are some of the reasons why I made z but that's personal whatever is chosen should be simple, I mean what do you think this github uses for example?

**Comment by Hussseinkizz** [comment]
> I checked the demo @Hussseinkizz built, and also checked Z.js documentation and reviews.
> 
> With full respect @Hussseinkizz, I think it should be disclosed to everyone that you are the primary author of Z.js, and it is yet to get a decent number of reviews and community support. If we have to resolve bugs with the framework we would all be mostly relying on just you. Unless @Buzzpy is ok with that dependency, I would suggest going with a more known and tested framework, like Astro.
> 
> I hope you don't mind me saying that, I'm in great admiration of the work you've done to build a framework, and I aspire to be able to do so myself.
> 
> On a separate note, I don't understand the logic of putting all the data in a single JSON file? As it grows, it will take longer and longer to load, so the website will be blank till it does. Whereas individual files will load quicker, if not all at once.

Thanks for these insights @RayMathew yes am currently the primary contributor and author of z js, and am really looking for contributors, it has so far took me 10 months to have it where it is, still long way to go, therefore, with that if you read my pr comments, you can see where I say my intention was not necessarily to have this project switch to z, but I made z for these kind of projects where you don't want too much of a framework and also not too less, so I wanted to demonstrate how z can chip in here that's if people on this project are willing to join the mess and contribute improvements and features to z, otherwise I think z will be prod ready at v1.0,0 cause its just v0.0.8 for now, still work in progress... 

On the issue of loading all files in a single file, I answered this already also I had hinted on it in my pr if one read closely.

Otherwise thanks for insights!

**Comment by Ibrahim-Faisal15** [comment]
> > So when should the transition to the framework should begin or it has already begun?
> 
> Just get hands on, do the thing in that framework, make a pr we all take a look, its already begun :)

Is it? Shouldn't the developer publish the ongoing work in newly-made branch, so that  we can see the ongoing process and take necessary steps?

**Comment by Hussseinkizz** [comment]
> > > So when should the transition to the framework should begin or it has already begun?
> > 
> > 
> > Just get hands on, do the thing in that framework, make a pr we all take a look, its already begun :)
> 
> Is it? Shouldn't the developer publish the ongoing work in newly-made branch, so that we can see the ongoing process and take necessary steps?

I meant one should do a significant foundation in a given direction for others to reason from and also chip in... this of course assumes best practices are followed!

**Comment by Buzzpy** [comment]
I really appreciate all your comments. I can't explain how important and supportive they are, being a backend Python developer maintaining a website like this can be difficult.

This issue has been going on for days and I think it's time to make a decision. I might lose really supportive, great contributors but for the sake of this project's users, **I've decided to go with the framework: Astro.**

Here's a complete report on why I'm choosing Astro for the project.

## Full Comparison: Z JS vs. Astro JS
### 1. Simplicity
- Z JS: Z JS is minimalistic and straightforward, with fewer built-in features. It's easy to start with and doesn't require much setup, making it easy for someone like me.
- Astro JS: Astro is also designed with simplicity in mind but offers more features. While it requires a bit more setup than Z JS, it still looks user-friendly, especially for those familiar with web development, unlike me. But that's fine, I have some great, and highly skilled contributors helping me out here, right?

Winner: Z JS (slightly simpler)

### 2. Performance (Speed and Accessibility)
- Z JS: Z JS is lightweight, which generally leads to fast load times. However, it might lack advanced performance optimization features, especially with large JSON files and all.
- Astro JS: Astro excels in performance by pre-rendering most content, leading to fast load times and improved accessibility. It's designed to be fast, with only the necessary JavaScript loaded, making it highly optimized for performance.
Winner: Astro JS (better performance optimization)

### 3. Ease of Maintenance
- Z JS: Z JS is easy to maintain for small projects due to its simplicity. However, as this project grows with more and more terms adding everyday, the lack of built-in structure and features might make maintenance more challenging.
Astro JS: Astro’s architecture makes it easier to manage and maintain when the project grows larger. It’s designed to keep code modular and organized, which will simplify (according to my understanding) updates and changes over time.
Winner: Astro JS (easier to maintain over time)

### 4. Documentation
Z JS: Z JS has less extensive documentation compared to Astro. This can be a hurdle especially for me where documentation is my only source of knowledge and I have no experience working with JS frameworks at all.
Astro JS: Astro has comprehensive and well-organized documentation. It’s actively maintained and covers a wide range of topics, making it easier for me to learn.
Winner: Astro JS (better documentation)

### 5. Reviews
Z JS: Z JS has a niche but loyal user base. It’s praised for its simplicity but might be seen as too basic for larger projects. Reviews are generally positive but highlight its limitations.
Astro JS: Astro has received excellent reviews, particularly for its simplicity as well as the performence. Developers seem to appreciate its balance between ease of use and powerful features.
Winner: Astro JS (stronger and more positive reviews)

**And the overall winner would be: AstroJS.**

Starting now, this project is accepting PRs integrating Astro with the project. I'll create a new branch in this repo named `astro-integration` and please contribute to that branch. Once we are completed with this integration, we will switch it to the main branch and deploy it on devpedia.pages.dev.

I promise, this will be the last time mentioning you guys, just wanted to make sure everyone's updated!

 @RayMathew @bartosz-skejcik @Brajo280603 @DanielAndresCP 

Appreciate your support, comments and everything!

**Comment by Ibrahim-Faisal15** [comment]
Gotta take a crash course on Astro.js first 😅

**Comment by Ibrahim-Faisal15** [comment]
> I really appreciate all your comments. I can't explain how important and supportive they are, being a backend Python developer maintaining a website like this can be difficult.
> 
> This issue has been going on for days and I think it's time to make a decision. I might lose really supportive, great contributors but for the sake of this project's users, **I've decided to go with the framework: Astro.**
> 
> Here's a complete report on why I'm choosing Astro for the project.
> 
> ## Full Comparison: Z JS vs. Astro JS
> ### 1. Simplicity
> * Z JS: Z JS is minimalistic and straightforward, with fewer built-in features. It's easy to start with and doesn't require much setup, making it easy for someone like me.
> * Astro JS: Astro is also designed with simplicity in mind but offers more features. While it requires a bit more setup than Z JS, it still looks user-friendly, especially for those familiar with web development, unlike me. But that's fine, I have some great, and highly skilled contributors helping me out here, right?
> 
> Winner: Z JS (slightly simpler)
> 
> ### 2. Performance (Speed and Accessibility)
> * Z JS: Z JS is lightweight, which generally leads to fast load times. However, it might lack advanced performance optimization features, especially with large JSON files and all.
> * Astro JS: Astro excels in performance by pre-rendering most content, leading to fast load times and improved accessibility. It's designed to be fast, with only the necessary JavaScript loaded, making it highly optimized for performance.
>   Winner: Astro JS (better performance optimization)
> 
> ### 3. Ease of Maintenance
> * Z JS: Z JS is easy to maintain for small projects due to its simplicity. However, as this project grows with more and more terms adding everyday, the lack of built-in structure and features might make maintenance more challenging.
>   Astro JS: Astro’s architecture makes it easier to manage and maintain when the project grows larger. It’s designed to keep code modular and organized, which will simplify (according to my understanding) updates and changes over time.
>   Winner: Astro JS (easier to maintain over time)
> 
> ### 4. Documentation
> Z JS: Z JS has less extensive documentation compared to Astro. This can be a hurdle especially for me where documentation is my only source of knowledge and I have no experience working with JS frameworks at all. Astro JS: Astro has comprehensive and well-organized documentation. It’s actively maintained and covers a wide range of topics, making it easier for me to learn. Winner: Astro JS (better documentation)
> 
> ### 5. Reviews
> Z JS: Z JS has a niche but loyal user base. It’s praised for its simplicity but might be seen as too basic for larger projects. Reviews are generally positive but highlight its limitations. Astro JS: Astro has received excellent reviews, particularly for its simplicity as well as the performence. Developers seem to appreciate its balance between ease of use and powerful features. Winner: Astro JS (stronger and more positive reviews)
> 
> **And the overall winner would be: AstroJS.**
> 
> Starting now, this project is accepting PRs integrating Astro with the project. I'll create a new branch in this repo named `astro-integration` and please contribute to that branch. Once we are completed with this integration, we will switch it to the main branch and deploy it on devpedia.pages.dev.
> 
> I promise, this will be the last time mentioning you guys, just wanted to make sure everyone's updated!
> 
> @RayMathew @bartosz-skejcik @Brajo280603 @DanielAndresCP @ManorSailor
> 
> Appreciate your support, comments and everything!

Gotta take a crash course on Astro.js first 😅

**Comment by Ibrahim-Faisal15** [comment]
Gotta take a quick crash course on Astro.ja first 😅

**Comment by Hussseinkizz** [comment]
I think this is right decision @Buzzpy Astro we go, me am still excited to see where this project goes, so am still here, will see how I also contribute with Astro it's ok, also it feels so right with all awesome people we have here who will lead us to it. Nice!

**Comment by Buzzpy** [comment]
Update: I built the base and set up the website with Astro, and I'm currently adding Markdown files, separately for each term with the help of ChatGPT. You can view the deployed site on: [https://astro-devpedia.pages.dev/](https://astro-devpedia.pages.dev/)

Some help in refactoring the code in the branch `astro-integration` is greatly appreciated!

**Comment by Hussseinkizz** [comment]
Hey, I see the astro branch, not bad, though on migrating the items, had you finished migrating all of them, if not, you can use the json file I did as I didn't leave any item out, and then you can just write a script that creates a file for such item and adds the content there in format you want, this also what I kinda did to get from a long HTML file to just a json file in brief time, also can the terms be categorized, under category directories in content, otherwise so far not bad at all.

**Comment by Ibrahim-Faisal15** [comment]
I highly agree on @Hussseinkizz proposal of using a JSON file for each card, as it can also avoid any  inconveniences faced later on.

**Comment by Buzzpy** [comment]
> Hey, I see the astro branch, not bad, though on migrating the items, had you finished migrating all of them, if not, you can use the json file I did as I didn't leave any item out, and then you can just write a script that creates a file for such item and adds the content there in format you want, this also what I kinda did to get from a long HTML file to just a json file in brief time, also can the terms be categorized, under category directories in content, otherwise so far not bad at all.

I see, and this approach looks easier. Can there be any downsides of using JSON files instead of Markdown files?

**Comment by ManorSailor** [comment]
Congrats on migrating to Astro :rocket:
I was busy with work this Saturday, so didn't have the time to look over here. 

Though, it appears that @Buzzpy has migrated to Astro. That's good. 

Although, it makes me wonder why are we still going with JSON files as opposed to Markdown files? Markdown are easier for writing content.

**Comment by Hussseinkizz** [comment]
> > Hey, I see the astro branch, not bad, though on migrating the items, had you finished migrating all of them, if not, you can use the json file I did as I didn't leave any item out, and then you can just write a script that creates a file for such item and adds the content there in format you want, this also what I kinda did to get from a long HTML file to just a json file in brief time, also can the terms be categorized, under category directories in content, otherwise so far not bad at all.
> 
> I see, and this approach looks easier. Can there be any downsides of using JSON files instead of Markdown files?
> 
> Update: It was way faster, I used a Python script to convert to JSON files and deployed it on [https://astro-devpedia.pages.dev](https://astro-devpedia.pages.dev/) !

Well I didn't mean we use json files, I was like it could be easier to have a script that reads from the json and outputs the markdown files for only the current migration purposes, hope that's clear!

**Comment by ManorSailor** [comment]
@Hussseinkizz 

I suppose I mixed up Z.js and Vanilla JS solution there a bit. I can't think of any reason why using separate json files could cause a problem.
PS: Apologies, I missed this comment before.

**Comment by Sudharshaun** [comment]
Great work @Buzzpy , though I was under the impression we were going to use MDX or MD for content generation instead of JSON. Wouldn't it be easier to integrate MD with the project as Astro.js provides out of box support for this..

**Comment by Buzzpy** [comment]
Hello there! I decided to go with JSON files for a few reasons:
- Structured Data: JSON is **designed** for structured data with key-value pairs while Markdown is for text content (but it's fine because Astro supports Markdown, but JSON still feels right in this case)
- Flexibility: JSON allows complex data structures like nested objects which might be useful when this project grows and we add multiple resources, keyword filtering, and all.
- Ease of use: JSON is easily parsed and manipulated in code, ideal for dynamic features. As for my understanding, Mardkwon is best for static content?
- Performance: I might be wrong with this, but JSON is quickly parsed by JavaScript,  while Markdown is pre-rendered. Since this website would go through changes a lot, JSON might be more suitable.

Feel free to correct me if I'm wrong, by the way 😊

**Comment by DOOMitru** [comment]
@Buzzpy If you do plan on using JSON files, I've already done the tedious work of splitting all the descriptions into their own individual json files. You can find them [here](https://github.com/DOOMitru/Dev-Encyclopedia/tree/chore/split-descriptions-into-cards/cards).

**Comment by Hussseinkizz** [comment]
> @Buzzpy If you do plan on using JSON files, I've already done the tedious work of splitting all the descriptions into their own individual json files. You can find them [here](https://github.com/DOOMitru/Dev-Encyclopedia/tree/chore/split-descriptions-into-cards/cards).

Nice @DOOMitru but don't you think separating these into separate json file per category or keyword is better than having just a single item per son json file? cause loading those ideally, requires you then to pick one by one and render them all? like whats the point? its not that we rendering one per time? the idea of finding a way to split them is cool, we just need the right level of splitting though, respectfully though I realize that work you did was very tedious!

**Comment by DOOMitru** [comment]
@Hussseinkizz I didn't separate the initial list of cards, but the information that pops up when you click the Explain button on each card. As for the work being tedious, it was only tedious for the first 10-15 and then I realized I could just write a script to do that for me. It took some tweaking but it was still faster than doing everything manually.

**Comment by Hussseinkizz** [comment]
> @Hussseinkizz I didn't separate the initial list of cards, but the information that pops up when you click the Explain button on each card. So instead of having to load all the details for all the cards, you could just load individual card details as needed. As for the work being tedious, it was only tedious for the first 10-15 and then I realized I could just write a script to do that for me. It took some tweaking but it was still faster than doing everything manually.

Ohh that sounds cool actually if you realized also in my first trial with z, I also used a script to move the stuff from the HTML to a json file, and now you took that step further I think that's good.. now are we going the HTML way? or I bring back my Z, I need contributors on it too :)

**Comment by Buzzpy** [comment]
### A heads up: #79  🚀
Once the Astro-integrated version is merged into the main branch, I'll be closing this issue. 

For any opinions or comments, use the Discussions! ✨

**Comment by KC900201** [comment]
> > > Hey, I see the astro branch, not bad, though on migrating the items, had you finished migrating all of them, if not, you can use the json file I did as I didn't leave any item out, and then you can just write a script that creates a file for such item and adds the content there in format you want, this also what I kinda did to get from a long HTML file to just a json file in brief time, also can the terms be categorized, under category directories in content, otherwise so far not bad at all.
> > 
> > 
> > I see, and this approach looks easier. Can there be any downsides of using JSON files instead of Markdown files?
> > Update: It was way faster, I used a Python script to convert to JSON files and deployed it on [https://astro-devpedia.pages.dev](https://astro-devpedia.pages.dev/) !
> 
> Well I didn't mean we use json files, I was like it could be easier to have a script that reads from the json and outputs the markdown files for only the current migration purposes, hope that's clear!

@Buzzpy, sorry but I tried to access https://astro-devpedia.pages.dev/ but the page seems down? Is the deployment progress still ongoing?

**Comment by Brajo280603** [comment]
@KC900201 The page is now deployed at devpedia.dev

please refer to #79

**Comment by RayMathew** [comment]
Shouldn't this issue be closed?
